R/bm-tpc-h.R

```diff
@@ -341,3 +358,41 @@ tpch_answer <- function(scale_factor, query_id, source = c("arrowbench", "duckdb
 
   answer
 }
+
+#' Get a SQL query
+#'
+#' Produces a function that can be queried against any DBI backend (e.g. DuckDB)
+#'
+#' @param con the connection to DuckDB to use to get the query (note: this is
+#' not the connection that will be used to run the query)
+#' @param query_num the query number to fetch the result for
+#'
+#' @return a function that accepts an argument `con` which will run
+#' `DBI::dbGetQuery()` against.
+#'
+#' @export
+#' @keywords internal
+get_sql_query_func <- function(con, query_num) {
+  query_sql <- get_sql_tpch_query(con, query_num)
+
+  # wrap the SQL in a function
+  function(input_func = NULL, collect_func = NULL, con) {
+    DBI::dbGetQuery(con, query_sql)
+  }
+}
+
+
+get_sql_tpch_query <- function(con, query_num) {
+  ensure_custom_duckdb()
+
```

Comment:
_nods_ agreed on the `con` front. I'm re-working the code to do like you suggest and create a new one for each. 

I'm a bit hesitant to grab the queries from github since that's fragile both with the location in their git repo, github being up, and needing an active internet connection (though the current install process relies on that as well, there should soon be a way to avoid that)

Reply:
Oh, I like that — though we could even pull them from duckdb directly in that script (it's a bit less of a concern of suddenly breaking / needing internet access, but I suspect that the duckdb tpch query interface is slightly more stable than the location of the files). Maybe we could add that when we look at #67 ?